modules/ROOT/pages/ai.adoc

```diff
@@ -0,0 +1,117 @@
+:description_short: Query AI APIs and accept results.
+:keywords: plugin, ai, artificial intelligence, query, search.
+:pluginname: AI Assistant
+:plugincode: ai
+:plugincategory: premium
+
+include::partial$misc/admon-ai-pricing.adoc[]
+
+include::partial$misc/admon-requires-6.6v.adoc[]
+
+The {pluginname} plugin allows a user to interact with registered AI APIs by sending queries and viewing responses within a {productname} editor dialog.
+
+Once a response is generated and displayed within the dialog, the user can choose to either:
+
+. insert it into the editor at the current selection;
+. create another query to further refine the response generated by the AI; or
+. close the dialog and discard the returned response.
+
+Users can retrieve a history of their conversations with the AI using the xref:#getThreadLog[`getThreadLog` API], including any discarded responses.
+
+[IMPORTANT]
+.On the absence of an {pluginname} demo
+====
+This initial release of the {pluginname} developer documentation does not include an in-page working demo.
+====
+
+== Basic setup
+
+To add the {pluginname} plugin to the editor, add both `{plugincode}` to the `plugins` option in the editor configuration and the `ai_request` function to the editor configuration.
+
+For example, interfacing with the OpenAI Completions API:
+
+[source,js]
+----
+const api_key = '<INSERT_API_KEY_HERE>';
+
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'ai',
+  toolbar: 'aidialog aishortcuts',
+  ai_request: (request, respondWith) => {
+    const openAiOptions = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Authorization': `Bearer ${api_key}`
+      },
+      body: JSON.stringify({
+        model: 'gpt-3.5-turbo',
+        temperature: 0.7,
+        max_tokens: 800,
+        messages: [{ role: 'user', content: request.prompt }],
+      })
+    };
+    respondWith.string((signal) => window.fetch('https://api.openai.com/v1/chat/completions', { signal, ...openAiOptions })
+      .then(async (response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
+          return Promise.reject(`Failed to communicate with the OpenAI API. ${await response.text()}`);
+        }
+      })
+      .then((data) =>
+        data.error
+        ? Promise.reject(`Failed to communicate with the ChatGPT API because of ${data.error.type} error: ${data.error.message}`)
+        : data)
+      .then((data) =>
+        // Extract the response content from the data returned by the API
+        data?.choices[0]?.message?.content?.trim()
+      )
+    );
+  }
+});
+----
+
+== Using a proxy server with {pluginname}
+
+A proxy server can provide flexibility by allowing extra processing before the request is sent to an LLM AI endpoint and before returning the response to the user.
```

Comment:
This is *not* why you should use a proxy service. You should use it because the [OpenAI Best Practices](https://help.openai.com/en/articles/5112595-best-practices-for-api-key-safety) make it very clear you should do so, and failing to do so exposes your organisation to financial and service uptime risks (when whoever walked off with your API key exhausts your quota).

Reply:
[Material added](https://github.com/tinymce/tinymce-docs/commit/8e2cc2f8c595f4d56549c0356d568374b3713388).